Utils/azuretools-core/src/com/microsoft/azure/toolkit/lib/common/task/AzureTaskManager.java

```diff
@@ -45,107 +45,211 @@ public static AzureTaskManager getInstance() {
         return AzureTaskManager.instance;
     }
 
-    public final Observable<Void> read(Runnable task) {
-        return this.read(new AzureTask<>(task));
+    public final void read(Runnable task) {
```

Comment:
We have many method overloads here, which could be a bit confused, maybe we could leverage patterns like builder to reduce the overloads later